test/plugin_functional/test_suites/core_plugins/rendering.ts

```diff
@@ -22,40 +22,41 @@ import expect from '@kbn/expect';
 import '../../plugins/core_provider_plugin/types';
 import { PluginFunctionalProviderContext } from '../../services';
 
+declare global {
+  interface Window {
+    __RENDERING_SESSION__: string[];
+  }
+}
+
 // eslint-disable-next-line import/no-default-export
 export default function({ getService, getPageObjects }: PluginFunctionalProviderContext) {
   const PageObjects = getPageObjects(['common']);
+  const appsMenu = getService('appsMenu');
   const browser = getService('browser');
   const find = getService('find');
   const testSubjects = getService('testSubjects');
 
-  function navigate(path: string) {
-    return browser.get(`${PageObjects.common.getHostPort()}${path}`);
-  }
-
-  function getLegacyMode() {
-    return browser.execute(() => {
+  const navigateTo = (path: string) =>
+    browser.navigateTo(`${PageObjects.common.getHostPort()}${path}`);
+  const getLegacyMode = () =>
+    browser.execute(() => {
       return JSON.parse(document.querySelector('kbn-injected-metadata')!.getAttribute('data')!)
         .legacyMode;
     });
-  }
-
-  function getUserSettings() {
-    return browser.execute(() => {
+  const getUserSettings = () =>
+    browser.execute(() => {
       return JSON.parse(document.querySelector('kbn-injected-metadata')!.getAttribute('data')!)
         .legacyMetadata.uiSettings.user;
     });
-  }
-
-  async function init() {
+  const exists = (selector: string) => testSubjects.exists(selector, { timeout: 2000 });
+  const init = async () => {
     const loading = await testSubjects.find('kbnLoadingMessage', 5000);
-
     return () => find.waitForElementStale(loading);
-  }
+  };
```

Comment:
Do we really need the double promise / double await ( `const loaded = await init()` + `await loaded()`) ?

Reply:
Yes, as we need to capture injected metadata before it is removed, but still need access to the element prior to that happening/